node/src/actors/chain_manager/mining.rs

```diff
@@ -73,92 +73,100 @@ impl ChainManager {
         // The highest checkpoint beacon should contain the current epoch
         beacon.checkpoint = current_epoch;
 
-        // TODO: Use a real PoE
-        let poe = true;
-        if poe {
-            // FIXME (tmpolaczyk): block creation must happen after data request mining
-            // (we must wait for all the potential nodes to sent their transactions)
-            // The best way would be to start mining a few seconds _before_ the epoch
-            // checkpoint, but for simplicity we just wait for 5 seconds after the checkpoint
-            ctx.run_later(Duration::from_secs(5), move |act, ctx| {
-                info!(
-                    "{} Discovered eligibility for mining a block for epoch #{}",
-                    Yellow.bold().paint("[Mining]"),
-                    Yellow.bold().paint(beacon.checkpoint.to_string())
-                );
-                // Send proof of eligibility to chain manager,
-                // which will construct and broadcast the block
-
-                act.create_tally_transactions()
-                    .join(
-                        signature_mngr::sign(&beacon)
-                            .map_err(|e| error!("Couldn't sign beacon: {}", e)),
+        // FIXME (tmpolaczyk): block creation must happen after data request mining
+        // (we must wait for all the potential nodes to sent their transactions)
+        // The best way would be to start mining a few seconds _before_ the epoch
+        // checkpoint, but for simplicity we just wait for 5 seconds after the checkpoint
+        ctx.run_later(Duration::from_secs(5), move |act, ctx| {
+            info!(
+                "{} Discovered eligibility for mining a block for epoch #{}",
+                Yellow.bold().paint("[Mining]"),
+                Yellow.bold().paint(beacon.checkpoint.to_string())
+            );
+            // Send proof of eligibility to chain manager,
+            // which will construct and broadcast the block
+            signature_mngr::vrf_prove(VrfMessage::block_mining(beacon))
```

Comment:
I would put a comment to specify that is need to check BlockEligibilityClaim before to create a block